### PR TITLE
PD-942: Add isHeader prop to table cell component

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

This PR adds an `isHeader` prop to the `Cell` component in the `@leafygreen-ui/table` package. This prop removes an outstanding barrier for docs to use our table component.

There were two main approaches which I considered taking for this API:
1. Make the existing TableHeader component work within Rows.
2. Add a prop to the Cell component to render a Cell as a `th` element.

I opted to implement the second option for a few reasons. The first is that the change is a significantly smaller one. Significant work would've needed to be done in TableHeader that would've resulted in far more code, overwriting quite a few styles, and would've duplicated a lot of code. This approach also makes the path to using TH elements in rows for existing Table implementations slightly simpler.

I also broke up the tests for Table by the constituent components. The tests were already structured by component, so while the changes look fairly large, almost nothing has changed within the tests aside from the addition of a test for this prop.

🎟 _Jira ticket:_ [Add support for Table Headers in Rows within Table](https://jira.mongodb.org/browse/PD-942)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/table/src/Cell.tsx [hunk 1]
```diff
@@ -5,30 +5,58 @@ import { commonCellStyles } from './styles';
 
 export const tdInnerDiv = createDataProp('td-inner-div');
 
-type CellProps = HTMLElementProps<'td', HTMLTableCellElement>;
+interface HeaderCellProps
+  extends HTMLElementProps<'th', HTMLTableHeaderCellElement> {
+  isHeader?: true;
```
Comment (brookescarlett) [inline packages/table/src/Cell.tsx:10]: Don't think this should be optional. isHeader is necessarily true for us to be using HeaderCellProps

File: packages/table/src/Cell.tsx [hunk 2]
```diff
@@ -5,30 +5,58 @@ import { commonCellStyles } from './styles';
...
+      props.scope = 'row';
+    }
+
     return (
-      <td
-        ref={ref}
-        className={cx(commonCellStyles, tdStyles, className)}
-        {...rest}
-      >
+      <Element {...props} {...rest}>
```
Comment (brookescarlett) [inline packages/table/src/Cell.tsx:55]: nit: Would prefer that we call this Root or Component to be consistent with other occasions across the repo where we do something similar, but this isn't imperative

Comment (DesignerDave) [reply packages/table/src/Cell.tsx]: Done

File: packages/table/src/Cell.tsx (hunk 1 above)
Comment (DesignerDave) [reply packages/table/src/Cell.tsx]: You're right

**Review by brookescarlett — APPROVED** [review]
LGTM

**Pull request merged**